mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping

```diff
@@ -0,0 +1,278 @@
+	METHOD m_syzhwirw bandlands ()Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;
+	METHOD m_toxcqvlp aboveY (Lnet/minecraft/unmapped/C_blhyfywp;I)Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		ARG 0 anchor
+		ARG 1 surfaceDepthMultiplier
+	METHOD m_ufzmgyjh steep ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+	METHOD m_uydhxaje temperature ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+	METHOD m_wfmmaumg not (Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;)Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		ARG 0 condition
+	METHOD m_zsodpnuj abovePreliminarySurface ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+	CLASS C_bxatslrt BlockMaterialRule
+		FIELD f_kmvgupfq resultState Lnet/minecraft/unmapped/C_txtbiemp;
+		FIELD f_vgrgkdrg rule Lnet/minecraft/unmapped/C_flergppe$C_yejovugv;
+		FIELD f_ybzzhjtf RULE_CODEC Lcom/mojang/serialization/Codec;
+		METHOD <init> (Lnet/minecraft/unmapped/C_txtbiemp;)V
+			ARG 1 resultState
+		METHOD <init> (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_flergppe$C_yejovugv;)V
+			ARG 1 resultState
+			ARG 2 rule
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+		METHOD equals (Ljava/lang/Object;)Z
+			ARG 1 o
+		METHOD m_cyuykmpj resultState ()Lnet/minecraft/unmapped/C_txtbiemp;
+		METHOD m_tdbowkom rule ()Lnet/minecraft/unmapped/C_flergppe$C_yejovugv;
+	CLASS C_dggoeuwn MaterialRule
+		FIELD f_dviqdsjz CODEC Lcom/mojang/serialization/Codec;
+		METHOD m_lffbnqbg bootstrap ()Lcom/mojang/serialization/Codec;
+		METHOD m_qnxeekwh codec ()Lcom/mojang/serialization/Codec;
+	CLASS C_dvxuzsoy SteepMaterialCondition
+		FIELD f_ojcqwxnd CONDITION_CODEC Lcom/mojang/serialization/Codec;
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+	CLASS C_dzzznwrr Context
+		FIELD f_cnbjadvz SURFACE_CELL_MASK I
+		FIELD f_wgttrquc SURFACE_CELL_BITS I
+		FIELD f_zstmpnis SURFACE_CELL_SIZE I
+	CLASS C_ebtxtbmd BiomeMaterialCondition
+		FIELD f_imktpjfv biomes Ljava/util/List;
+		FIELD f_khqxzpno CONDITION_CODEC Lcom/mojang/serialization/Codec;
+		METHOD <init> (Ljava/util/List;)V
+			ARG 1 biomes
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+		METHOD equals (Ljava/lang/Object;)Z
+			ARG 1 o
+		METHOD m_owocisww biomes ()Ljava/util/List;
+	CLASS C_hubqzhng ConditionMaterialRule
+		FIELD f_kejiyqxr thenRun Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;
+		FIELD f_mniixoud RULE_CODEC Lcom/mojang/serialization/Codec;
+		FIELD f_uhernjdb ifTrue Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		METHOD <init> (Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;)V
+			ARG 1 ifTrue
+			ARG 2 thenRun
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+		METHOD equals (Ljava/lang/Object;)Z
+			ARG 1 o
+		METHOD m_kdsbbibg ifTrue ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		METHOD m_szdsyqjr thenRun ()Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;
```

Comment:
thenRun? Idk why but this feels off. Ignore if it is meant to be like this.

Reply:
This matches the codec field